File: frontend/src/app/dto/order/order-response.ts
export interface Order {
    orderId: number;
    userId: number;
    items: OrderItem[];
    status: OrderStatus;
    totalAmount: number;
    createdAt: Date;
    recipientName: string;
    contactEmail: string;
    shippingAddress: string;
    contactPhone: string;

  }
  
  export interface OrderItem {
    id: number;
    menuItemId: number;
    name: string;
    quantity: number;
    price: number;
  }
  
  export enum OrderStatus {
    PENDING = 'PENDING',
    PREPARING = 'PREPARING',
    READY = 'READY',
    DELIVERED = 'DELIVERED',
    CANCELLED = 'CANCELLED'
  }

File: order-service/src/main/java/com/devsoncall/streammymeal/order/dto/OrderResponse.java
package com.devsoncall.streammymeal.order.dto;

import jakarta.persistence.Column;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class OrderResponse {
    private Integer orderId;
    private Integer userId;
    private LocalDateTime orderDate;
    private BigDecimal totalAmount;
    private String status;
    private String paymentStatus;
    private String recipientName;
    private String contactEmail;
    private String shippingAddress;
    private String contactPhone;
    private List<OrderItemResponse> items;

}